packages/mdc-grid-list/README.md

```diff
@@ -0,0 +1,195 @@
+        <img class="mdc-grid-tile__primary-content" src="images/1-1.jpg" />
+      </div>
+      <span class="mdc-grid-tile__secondary">
+        <span class="mdc-grid-tile__title">Title</span>
+        <span class="mdc-grid-tile__support-text">Support text</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+### Add icon to secondary content (caption)
+
+You can add icon to caption by adding `mdc-grid-list--with-icon-align-start` or
+`mdc-grid-list--with-icon-align-end` and changing the markup.
+
+```html
+<div class="mdc-grid-list mdc-grid-list--with-icon-align-start">
+<!-- or aspect ratio you name it. -->
+  <ul class="mdc-grid-list__tiles">
+    <li class="mdc-grid-tile">
+      <div class="mdc-grid-tile__primary">
+        <img class="mdc-grid-tile__primary-content" src="images/1-1.jpg" />
+      </div>
+      <span class="mdc-grid-tile__secondary">
+        <i class="mdc-grid-tile__icon material-icons">star_border</i>
+        <span class="mdc-grid-tile__title">Title</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+### Use div in replace of img
+
+In case you cannot ensure all your assets will have the same aspect ratio, you
+can use `div` instead of `img` markup. It will resize the assets to cover the tile
+and crop the assets to display the center part.
+
+```html
+<div class="mdc-grid-list mdc-grid-list--header-caption">
+  <ul class="mdc-grid-list__tiles">
+    <li class="mdc-grid-tile">
+      <div class="mdc-grid-tile__primary">
+        <div class="mdc-grid-tile__primary-content demo-grid-tile-content"></div>
+      </div>
+      <span class="mdc-grid-tile__secondary">
+        <span class="mdc-grid-tile__title">Title</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+In this case, you should set `background-image:url(...)` instead of `src`. However,
+the method results in a less semantic markup, so we don't use this method by
+default.
+
+
+### RTL Support
+
+`mdc-grid-list` is automatically RTL-aware, and will re-position elements whenever
+it, or its ancestors, has a `dir="rtl"` attribute.
+
+
+### Theme
+
+`mdc-grid-list` support theme. Tile `mdc-grid-tile__primary` uses background and text on
+background, `mdc-grid-tile__secondary` uses the primary color and text on primary.
+
+
+### Using the Foundation Class
+
+MDCGridList ships with an `MDCGridListFoundation` class that external frameworks and libraries
+can use to build their own MDCGridList components with minimal effort. As with all foundation
+classes, an adapter object must be provided. The adapter for grid list must provide the following
+functions, with correct signatures:
+
+| Method Signature | Description |
+| --- | --- |
+| `getOffsetWidth() => number` | Get root element `mdc-grid-list` offsetWidth. |
+| `getTileOffsetWidthAtIndex(index: number) => number` | Get offsetWidth of `mdc-grid-tile` at specified index. |
+| `setTilesWidth(value: number) => void` | Set `mdc-grid-list__tiles` container width to value. |
```

Comment:
1. I agree that an adapter method that sets an arbitrary style is more generic than one which is supposed to set the width of an element. However, I'd argue that it's _less_ ambiguous, because setting a style property to a specific value is a completely unambiguous action for the DOM. Whereas "setting the width of the tiles element" is more ambiguous because _how_ the width is set is not immediately clear. Also, as you mentioned, using a `setStyle...` method name makes the adapter more flexible if additional style properties need to be updated in the future. 

2. I actually like **Scenario B**. The reason is because it is completely unambiguous. In this case, the frameworks are not setting the "width" of an element but the _width style property_, which should always accept a string. I feel that designing our adapter this way will take all of the guesswork out of what the adapter method is doing and how it should do it. Furthermore, using numbers alone, it may not be immediately clear what the unit is. For example, what if someone interprets the number as a percentage? Or as a `vh`/`vw` value? Whereas with a string, its unit is always attached to it (e.g. `20px`, `40vh`, etc.). This means that if we were ever to change the _unit_ we'd want to use to set the width, it would still work. Whereas with scenario A we'd be violating an adapter assumption about what the unit of measurement should be, and therefore it would be a breaking change. WDYT?